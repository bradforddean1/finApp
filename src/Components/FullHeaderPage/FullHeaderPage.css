.FullHeaderPage {
	height: 100%;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	background-repeat: no-repeat;
	background-size: 100%;
}

.FullHeaderPage h1 {
	text-align: center;
	margin-bottom: 20px;
}

.FullHeaderPage .slogan {
	font-family: Futura;
	font-weight: 500;
	font-size: 24px;
	line-height: 32px;
	color: #46d16d;
}
