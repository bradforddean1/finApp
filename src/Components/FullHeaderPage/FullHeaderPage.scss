.FullHeaderPage {
	height: 100%;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	background-repeat: no-repeat;
	background-size: 100% 176px;
}

.FullHeaderPage h1 {
	text-align: center;
	margin-bottom: 20px;
}

.FullHeaderPage .slogan {
	margin-right: 20px;
	font-family: "Hind", sans-serif;
	font-weight: 500;
	font-size: 24px;
	line-height: 32px;
	color: #46d16d;
}

.slogan {
	padding-left: 20px;
}
