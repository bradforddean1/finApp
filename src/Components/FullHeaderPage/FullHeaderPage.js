import React from "react";
import PropTypes from "prop-types";
import headerbg from "../../assets/header.svg";
import logo from "../../assets/ztok.svg";
import "./FullHeaderPage.css";

/**
 * Page outline with full branded header.
 * @component
 */
function FullHeaderPage(props) {
	const { children } = props;
	return (
		<div
			className="FullHeaderPage page"
			style={{ backgroundImage: `url(${headerbg})` }}
		>
			<div>
				<h1>
					<img src={logo} alt="ztok" />
				</h1>
				<span className="slogan">Stocks Under Control!</span>
			</div>

			{children}
		</div>
	);
}

FullHeaderPage.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.arrayOf(PropTypes.element),
	]),
};

export default FullHeaderPage;
