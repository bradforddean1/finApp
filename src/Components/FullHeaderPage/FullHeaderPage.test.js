import React from "react";
import ReactDOM from "react-dom";
import FullHeaderPage from "./FullHeaderPage";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("FullHeaderPage Component", () => {
	it("Renders", () => {
		const div = document.createElement("div");
		ReactDOM.render(<FullHeaderPage />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("Renders the Default State", () => {
		const wrapper = shallow(<FullHeaderPage />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("Renders given props", () => {
		const wrapper = shallow(
			<FullHeaderPage>
				<div>Test Page</div>
			</FullHeaderPage>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
