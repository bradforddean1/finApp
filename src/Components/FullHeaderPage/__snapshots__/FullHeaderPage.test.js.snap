// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullHeaderPage Component Renders given props 1`] = `
<div
  className="FullHeaderPage page"
  style={
    Object {
      "backgroundImage": "url(header.svg)",
    }
  }
>
  <div>
    <h1>
      <img
        alt="ztok"
        src="ztok.svg"
      />
    </h1>
    <span
      className="slogan"
    >
      Stocks Under Control!
    </span>
  </div>
  <div>
    Test Page
  </div>
</div>
`;

exports[`FullHeaderPage Component Renders the Default State 1`] = `
<div
  className="FullHeaderPage page"
  style={
    Object {
      "backgroundImage": "url(header.svg)",
    }
  }
>
  <div>
    <h1>
      <img
        alt="ztok"
        src="ztok.svg"
      />
    </h1>
    <span
      className="slogan"
    >
      Stocks Under Control!
    </span>
  </div>
</div>
`;
